src/paddle.cpp

```diff
@@ -163,14 +173,32 @@ void Paddle::runEffects() {
 	}
 
 	if (magnetTimer <= 0) {
-		magnetTimer = 2.0f;
-		magnetTimerStarted = false;
+		audio->playCue(HIT_CUE);
+		magnetBall->setMagnetised(false);
 
-		msgPtr = new Message(messageNS::MAGNET_EFFECT, messageNS::UNBIND, id, magnetBallId);
-		msgPtrTwo = new Message(messageNS::END_EFFECT, messageNS::BALL, effectNS::MAGNET, id);
+		magnetTimer = 1.0f;
+		magnetInitialized = false;
+		magnetised = false;
+		magnetBall = nullptr;
+	}
+}
 
-		pushMsg(msgPtr);
-		pushMsg(msgPtrTwo);
+void Paddle::initMagnetEffect(Entity &ent) {
+	audio->playCue(HIT_CUE);
+	magnetBall = (Ball*)&ent;
+	magnetBall->setVelocity(VECTOR2(0, 0));
+	magnetBall->setMagnetised(true);
+	magnetInitialized = true;
+	
+	int ballWidth = magnetBall->getWidth() * magnetBall->getScaleX();
+	int paddleWidth = spriteData.width * spriteData.scale.x;
+
+	// Ball will always stick to the front of paddle
+	if (magnetBall->getX() + ballWidth > spriteData.x && side == paddleNS::RIGHT) {
+		magnetBall->setX(spriteData.x - paddleWidth);
+	} 
+	else if (magnetBall->getX() - ballWidth < spriteData.x + paddleWidth && side == paddleNS::LEFT) {
+		magnetBall->setX(spriteData.x + paddleWidth);
```

Comment:
If im not wrong what this does is, when the ball sticks to the top/bottom of the paddle, it gets shifted to the corner of the paddle?